sdk/messaging/azservicebus/message.go

```diff
@@ -119,29 +119,34 @@ func (m *Message) toAMQPMessage() *amqp.Message {
 
 	// TODO: I don't think this should be strictly required. Need to
 	// look into why it won't send properly without one.
-	var messageID = m.MessageID
+	var messageID string
 
-	if messageID == "" {
+	if m.MessageID == nil || *m.MessageID == "" {
 		uuid, err := uuid.New()
 
 		if err == nil {
```

Comment:
What happens if `err != nil`?

Reply:
I actually think I can get rid of this workaround entirely now. I'll try it out real quick.

Reply:
Ah, I was! There were just a few little spots, but it works great. The ID is still auto-generated (we're not doing it anymore so it's just happening on the Service Bus side).

There was one spot I was concerned about (the RPC Link) but we're auto-generating IDs in there separately so we should be good.